### PR TITLE
Feature/cax 1 prs@158 Run smoke tests against consumers in PRS Smoke tests workflow.

### PR DESCRIPTION
**Pull request by ouphi**
Run smoke tests against consumers
- in dev006 environment
- in int environment

### CODE REVIEW

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java [hunk 1]
```diff
@@ -31,15 +31,23 @@
     private static final String SAMPLE_VIN = "YS3DD78N4X7055320";
     private static final String VIN = "vin";
     private static final String VIEW = "view";
+    String userName;
+    String password;
 
     @BeforeEach
     public void setUp() {
-        RestAssured.baseURI = "https://catenaxdev001akssrv.germanywestcentral.cloudapp.azure.com";
+        RestAssured.baseURI = userName = System.getProperty("baseURI");
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:39]: why are you assigning baseURI property to userName?

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java [hunk 2]
```diff
@@ -31,15 +31,23 @@
     private static final String SAMPLE_VIN = "YS3DD78N4X7055320";
     private static final String VIN = "vin";
     private static final String VIEW = "view";
+    String userName;
```
Comment (Izzzu) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:34]: can userName and password be private?

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java (hunk 1 above)
Comment (ouphi) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java]: Oops good catch thanks.

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java (hunk 2 above)
Comment (ouphi) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java]: Yes :).

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java [hunk 3]
```diff
@@ -31,15 +31,23 @@
...
+        RestAssured.baseURI = System.getProperty("baseURI");
+        userName = System.getProperty("userName");
+        password = System.getProperty("password");
     }
 
     @Test
     public void getPartsTreeByVin_success() {
-        given()
+        // Add basic auth if a userName and password have been specified.
+        var requestSpecification = (userName != null && password != null) ?
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:47]: A bit more elegant way is to build a rest assured `RequestSpecification` here (there is a `RequestSpecBuilder` for this) with the authentication configuration depending on the inputs. Then further down you can just do:
`given()
      .spec(requestSpecification)
      .pathParams(VIN, SAMPLE_VIN)
.when()
`

Comment (ouphi) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java]: Thanks! I did that.

**Review by marcgs — APPROVED** [review]

File: .github/workflows/partsrelationshipservice-smoketest.yml [hunk 4]
```diff
@@ -33,6 +37,11 @@ jobs:
...
-      - name: Build with Maven
-        run: |
-          mvn --batch-mode --update-snapshots -Dgroups=SmokeTests test
+      - name: Run PRS smoke tests
+        run: mvn --batch-mode --update-snapshots -Dgroups=SmokeTests test -DbaseURI=$PRS_BASEURI
+
+      - name: Run PRS Smoke tests against Kaputt connector in dev006 environment
+        run: mvn --batch-mode --update-snapshots -Dgroups=SmokeTests test -DbaseURI=$KAPUTT_DEV_CONSUMER_ARTIFACT_URI -DuserName=${{ secrets.KAPUTT_DEV_CONSUMER_USERNAME }} -Dpassword=${{ secrets.KAPUTT_DEV_CONSUMER_PASSWORD }}
+
+      - name: Run PRS Smoke tests against Kaputt connector in integration environment
```
Comment (marcgs) [inline .github/workflows/partsrelationshipservice-smoketest.yml:46]: I think we should either run smoke tests for dev (default) or int, but not both. This should be configurable with an input param. We can do this improvement in a separate PR.

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java [hunk 5]
```diff
@@ -31,15 +33,33 @@
...
     private static final String VIN = "vin";
     private static final String VIEW = "view";
+    private String userName;
+    private String password;
 
     @BeforeEach
     public void setUp() {
-        RestAssured.baseURI = "https://catenaxdev001akssrv.germanywestcentral.cloudapp.azure.com";
+        RestAssured.baseURI = System.getProperty("baseURI");
+        userName = System.getProperty("userName");
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:42]: How this will work when we run it on local ? Do we have instruction for that somewhere ?
Just want to understand.

Comment (ouphi) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:42]: Hmm good point I will document it and add a value by default.

**Review by Izzzu — APPROVED** [review]

**Review by cpeeyush — APPROVED** [review]

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java (hunk 5 above)
Comment (ouphi) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/smoketest/SmokeTests.java:44]: Resolving the conversation as I added the instruction in the readme. Need to add VM options to add username password and url

File: .github/workflows/partsrelationshipservice-smoketest.yml (hunk 4 above)
Comment (ouphi) [reply .github/workflows/partsrelationshipservice-smoketest.yml:46]: Why not? If we deploy in dev and int when we merge I think having smoke tests in both environments after the deploy workflow makes sense.

Comment (ouphi) [reply .github/workflows/partsrelationshipservice-smoketest.yml:46]: We can discuss this later. I will merge in the meantime.

Comment (marcgs) [reply .github/workflows/partsrelationshipservice-smoketest.yml:46]: We don't deploy to int automatically after a merge. See https://confluence.catena-x.net/display/CXM/PRS+Environments+and+Test+Data.
